src/renderers/shared/fiber/ReactFiberDevToolsHook.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberDevToolsHook
+ * @flow
+ */
+
+/* globals __REACT_DEVTOOLS_GLOBAL_HOOK__ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+import type { FiberRoot } from 'ReactFiberRoot';
+
+var warning = require('warning');
+
+let rendererID = null;
+if (
+  typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ !== 'undefined' &&
+  typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.supportsFiber
+) {
+  exports.injectInternals = function(internals : Object) {
+    warning(rendererID == null, 'Cannot inject into DevTools twice.');
+    rendererID = __REACT_DEVTOOLS_GLOBAL_HOOK__.inject({
+      ...internals,
+      // Increment when internals become incompatible with DevTools.
+      // This way older DevTools can ignore newer Fiber versions.
+      version: 1,
```

Comment:
Another example of incompatibility is changing types of work enum. If old tools can't just bail they might display everything incorrectly and it wouldn't be obvious they need to be updated.